lib/bolt/config.rb

```diff
@@ -387,5 +423,95 @@ def casefold(path)
         l =~ /[A-Za-z]/ ? "[#{l.upcase}#{l.downcase}]" : l
       end.join
     end
+
+    def config_loaded
+      msg = <<~MSG
+        CONFIGURATION FILES
```

Comment:
Okay I understand the concern about verbosity. I don't think we need to explain _how_ config is merged here, only that we're loading it from multiple places. The idea is to address users' confusion about where values that aren't in their `bolt.yaml` came from (that is, _discovering_ that there are multiple config files at play). Understanding how the final merged config was actually generated is something to consult the documentation for.

I was imagining something like:

```rb
@logger.debug("Loaded config from [...], [...]")
```